### ISSUE TITLE
Tmp folder with block files is being created when trying to export/run queries (8M rows)

### ISSUE THREAD

**Issue by drtransformacaostefan** [duckdb/duckdb#11605]
### What happens?

Since 09/04/2024 my query is no longer exporting with the same time is was before 09/04/2024. The query was taking less than 3 minutes to export 8 million rows to a parquet file and now its reaching 10 minutos and still no finishing. I am stopping manually causa don't want to keep forcing my machine

### To Reproduce

I can't share the data, but I will try to explain every step: first I create the file database by creating a connection to the .db file with some configurations like memory_limit to 3GB and threads to 2;

Then I create my base table with all the respective types for each column (41 columns total) and insert the data from two CSV files (first two weeks of February and last two weeks of February) this give me 21 million rows;

Next I create a zip code range table (4 columns) and insert data from xlsx file, about 10 thousand rows;

Finally I do the current query

```python
conn.execute("""COPY (
        SELECT BC.*
        , CR.state
        , CR.city
        FROM base_completa AS BC 
          LEFT JOIN cep_range AS CR 
            ON (BC.zip_code BETWEEN CR.zip_start AND CR.zip_end) 
        WHERE CR.state = 'SP'
             ) TO "finaloutput_csv.csv" (HEADER, DELIMITER ";")""")
```

And with the state filter the output was about 8 million rows

### OS:

Windows x64

### DuckDB Version:

0.10.1

### DuckDB Client:

Python

### Full Name:

Stefan Esidio

### Affiliation:

DR

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

No - I cannot share the data sets because they are confidential

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by drtransformacaostefan** [comment]
What example can I send to help you reproduce this issue? Cause my data is confidential, so I'd like to know what else can I do to help solve this

**Comment by drtransformacaostefan** [comment]
Good morning Mytherin, I managed to create the fake dataset to exemplify the error. I took the liberty of using the email you provide in your account to share the OneDrive [link](https://drinova-my.sharepoint.com/:f:/g/personal/stefan_drtransformacaodigital_com_br/Eguxn4U_NehLhKL2aS-3nfUBfMfEKLHhhgLqx8p7kJ0d6g) with the folder containing the files needed for testing as well as the jupyter notebook file with the step by step instructions.
When executing the base export, duckdb is creating a folder with more than 10GB of ".block" files. I hope that with these files it will be possible to reproduce the error!
